File: Commons/Data/src/main/java/com/greentree/commons/data/resource/location/IOResourceLocation.java
package com.greentree.commons.data.resource.location;

import com.greentree.commons.data.resource.IOResource;

public interface IOResourceLocation extends ResourceLocation {
	
	@Override
	IOResource getResource(String name);
	
	
	IOResource createNewResource(String name);
	
	default boolean deleteResource(String name) {
		final var res = getResource(name);
		if(res == null)
			return false;
		return res.delete();
	}
	
	default IOResource createResource(String name) {
		if(isExist(name))
			deleteResource(name);
		return createResource(name);
	}
	
}

File: Commons/Data/src/main/java/com/greentree/commons/data/resource/location/IOResourceLocation.kt
package com.greentree.commons.data.resource.location

import com.greentree.commons.data.resource.IOResource

interface IOResourceLocation : ResourceLocation {

	override fun getResource(name: String): IOResource
	fun createNewResource(name: String): IOResource
	fun deleteResource(name: String): Boolean {
		val res = getResource(name)
		return res.delete()
	}

	fun createResource(name: String): IOResource {
		if(isExist(name)) deleteResource(name)
		return createResource(name)
	}
}